### PR TITLE
modules/SceMotion: stub SceMotionGetState with set value to zero.

### PR DESCRIPTION
**Pull request by Zangetsu38**
- modules/SceMotion: stub SceMotionGetState with default values.
Fixes head movement of Sackboy occurring independently in LBP.
Also fixes the issue in Gravity Rush where the screen turns black when using gravity or during specific moments involving the motion sensor.
- motion: Prepares for future motion support.

# Result
- looking camera without moving
image

- FINALY fix gravity rush, move to playable state
image

### CODE REVIEW

File: vita3k/modules/SceDriverUser/SceMotion.cpp [hunk 1]
```diff
@@ -48,28 +51,76 @@ EXPORT(int, sceMotionGetDeadbandExt) {
...
 }
 
-EXPORT(int, sceMotionGetState) {
-    TRACY_FUNC(sceMotionGetState);
+EXPORT(int, sceMotionGetState, SceMotionState *motionState) {
+    TRACY_FUNC(sceMotionGetState, motionState);
+    if (!motionState)
+        return RET_ERROR(SCE_MOTION_ERROR_NULL_PARAMETER);
+
+    STUBBED("Set some default values");
```
Comment (Macdu) [inline vita3k/modules/SceDriverUser/SceMotion.cpp:79]:
```suggestion
    STUBBED("Set to zero");
```

File: vita3k/motion/include/motion/state.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+#pragma once
+
+#include <motion/motion.h>
+
+#include <SDL_gamecontroller.h>
+
+struct _SDL_GameController;
+
+typedef std::shared_ptr<_SDL_GameController> GameControllerPtr;
```
Comment (Macdu) [inline vita3k/motion/include/motion/state.h:26]: remove

File: gen-windows.bat [hunk 3]
```diff
@@ -1,4 +1,5 @@
 @echo off
+title Generate project solution VS
```
Comment (Macdu) [inline gen-windows.bat:2]: remove, not part of this PR

File: vita3k/motion/include/motion/state.h (hunk 2 above)
Comment (Zangetsu38) [reply vita3k/motion/include/motion/state.h:26]: hmm is used for motion suport in DS/4, is just for prepare next works

File: gen-windows.bat (hunk 3 above)
Comment (Zangetsu38) [reply gen-windows.bat:2]: is small change, i no like create pr for very small things like that is not important

File: vita3k/modules/SceDriverUser/SceMotion.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/modules/SceDriverUser/SceMotion.cpp:79]: i no set all to zero, is more default value in default position

File: vita3k/motion/include/motion/state.h (hunk 2 above)
Comment (Macdu) [reply vita3k/motion/include/motion/state.h:26]: Something starting with an underscore is definitely not meant to be used, remove

Comment (Zangetsu38) [reply vita3k/motion/include/motion/state.h:26]: ??? this line is litteraly same inside ctrl ? 🤔

Comment (Zangetsu38) [reply vita3k/motion/include/motion/state.h:26]: image

Comment (Macdu) [reply vita3k/motion/include/motion/state.h:26]: Ah ok, the c typedef issue. Anyway, no need to add this now and we may have to include Ctrl state which defines it anyway

Comment (Zangetsu38) [reply vita3k/motion/include/motion/state.h:26]: no need include ctrl state, is full diferent; only need this typdef, no any other things inside ctrl, so is no sense add full moiule for one things

Comment (Macdu) [reply vita3k/motion/include/motion/state.h:26]: Either you use it, either you do not add these lines

Comment (Zangetsu38) [reply vita3k/motion/include/motion/state.h]: done, but no removed, just disable for now, like pr said is stuff for prepare futur implementation, like that when start it juste enable again

**Comment by ArbestRiAgain** [comment]
FINALLY! GRAVITY RUSH NO LONGER SOFTLOCKS AFTER THAT ONE PART!

File: vita3k/motion/include/motion/state.h [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+#include <motion/motion.h>
+
+/** TODO: Enable this when start implementing motion support with SDL2
+#include <SDL_gamecontroller.h>
+
+// struct _SDL_GameController;
+
+// typedef std::shared_ptr<_SDL_GameController> GameControllerPtr;
+**/
```
Comment (Macdu) [inline vita3k/motion/include/motion/state.h:28]: What's the point of putting this, it'll most likely get deleted when correctly implementing it for many reasons. Remove.

Comment (Zangetsu38) [reply vita3k/motion/include/motion/state.h]: done

**Review by Macdu — APPROVED** [review]
